src/Features/CSharp/Portable/Copilot/CSharpCopilotNotImplementedMethodFixProvider.cs

```diff
@@ -0,0 +1,651 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Copilot;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.MethodImplementation;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.Copilot;
+
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.ImplementNotImplementedException), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed partial class CSharpCopilotNotImplementedMethodFixProvider() : SyntaxEditorBasedCodeFixProvider
+{
+    public override ImmutableArray<string> FixableDiagnosticIds { get; }
+        = [IDEDiagnosticIds.CopilotImplementNotImplementedExceptionDiagnosticId];
+
+    public override Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        RegisterCodeFix(context, CSharpAnalyzersResources.Implement_with_Copilot, nameof(CSharpAnalyzersResources.Implement_with_Copilot));
+        return Task.CompletedTask;
+    }
+
+    protected override async Task FixAllAsync(
+        Document document, ImmutableArray<Diagnostic> diagnostics,
+        SyntaxEditor editor, CancellationToken cancellationToken)
+    {
+        foreach (var diagnostic in diagnostics)
+            await FixOneAsync(editor, document, diagnostic, cancellationToken).ConfigureAwait(false);
+    }
+
+    private static async Task FixOneAsync(
+    SyntaxEditor editor, Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+    {
+        // Find the throw statement node
+        var throwStatement = diagnostic.AdditionalLocations[0].FindNode(getInnermostNodeForTie: true, cancellationToken).AncestorsAndSelf().OfType<StatementSyntax>().FirstOrDefault();
+        if (throwStatement == null)
+        {
+            return;
+        }
+
+        // Find the containing method
+        var containingMethod = throwStatement.AncestorsAndSelf().OfType<MethodDeclarationSyntax>().FirstOrDefault();
+        if (containingMethod == null)
+        {
+            return;
+        }
+
+        var containingClass = containingMethod.AncestorsAndSelf().OfType<ClassDeclarationSyntax>().FirstOrDefault();
+        // TODO: handle containingRecord and containingStruct, do not return if method is in a record.
+
+        if (containingClass == null)
+        {
+            return;
+        }
+
+        var containingMethodName = containingMethod.Identifier.Text;
+        var referencedMethods = new List<string>();
+
+        // Traverse the syntax tree to find all method invocations
+        var root = containingMethod.SyntaxTree.GetRoot(cancellationToken);
+        var methodInvocations = root.DescendantNodes().OfType<InvocationExpressionSyntax>();
```

Comment:
this is massively expensie.  what are you trying to do here?  if you are trying to find callers of the containing symbol, you should use the Find-Refs api.  It can take a list of documents to only produce references within.